### ISSUE TITLE
Pop up an alert if they try to deactivate any of our "must have" plugins

### ISSUE THREAD

**Issue by blogtutor** [blogtutor/blog-tutor-support#14]
Or at least put a dashboard notice if any of our plugins are deactivated.

- WP Rocket
- UpdraftPlus
- Sucuri Security

If it's a sticky notice, we might need a way to override some (in the settings page), since not all sites use all plugins. (Some are still using W3 Total Cache, and at least one doesn't want caching at all.)

**Comment by ecotechie** [comment]
Looks like WP Rocket does something like this:
https://github.com/wp-media/wp-rocket/blob/05b73c07f3b173c21970c526f2492b3fca114bcf/assets/js/wpr-modal.js#L55

**Comment by blogtutor** [comment]
Yep. I've seen a nearly identical popup on other plugins, too... I'm wondering if that's a third-party extension or maybe something put up on Github somewhere?

**Comment by ecotechie** [comment]
How about we just disable the ability all together for some plugins?
We can edit this to go into our plugin and to only allow it for us... 

add_filter( 'plugin_action_links', 'disable_plugin_deactivation', 10, 4 );
function disable_plugin_deactivation( $actions, $plugin_file, $plugin_data, $context ) {
	// Remove edit link for all plugins
	if ( array_key_exists( 'edit', $actions ) )
		unset( $actions['edit'] );
	// Remove deactivate link for important plugins
	if ( array_key_exists( 'deactivate', $actions ) && in_array( $plugin_file, array(
		'wpmu-dev-plusone/plusone.php',
		'plugin-folder-name/plugin.php',
                'plugin-folder-name/plugin.php'
	)))
		unset( $actions['deactivate'] );
	return $actions;
}

**Comment by blogtutor** [comment]
Ooh, that's sneaky!  I generally don't like to take "control" of someone's site away from them. They could always deactivate via FTP (but most folks won't know how)... let me think on this, philosophically.
